lib/plugin.js

```diff
@@ -0,0 +1,287 @@
+/*
+ * decaffeinate suggestions:
+ * DS101: Remove unnecessary use of Array.from
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/main/docs/suggestions.md
+ */
+// The plugin module manages the dynamic retrieval of plugin
+// javascript including additional scripts that may be requested.
+
+// forward = require './forward'
+
+let plugin;
+module.exports = (plugin = {});
+
+const escape = s => (''+s)
+  .replace(/&/g, '&amp;')
+  .replace(/</g, '&lt;')
+  .replace(/>/g, '&gt;')
+  .replace(/"/g, '&quot;')
+  .replace(/'/g, '&#x27;')
+  .replace(/\//g,'&#x2F;');
+
+// define loadScript that allows fetching a script.
+// see example in http://api.jquery.com/jQuery.getScript/
+
+const loadScript = function(url, options) {
+  console.log("loading url:", url);
+  options = $.extend(options || {}, {
+    dataType: "script",
+    cache: true,
+    url
+  }
+  );
+  return $.ajax(options);
+};
+
+const scripts = [];
+const loadingScripts = {};
+const getScript = (plugin.getScript = async function(url, callback) {
+  if (callback == null) { callback = function() {}; }
+  if (Array.from(scripts).includes(url)) {
+    return callback();
+  } else {
+    await new Promise(function(resolve, reject) {
+      loadScript(url)
+        .done(function() {
+          scripts.push(url);
+          return resolve();
+        }).fail(function(_jqXHR, _textStatus, err) {
+          console.log('getScript: Failed to load:', url, err);
+          return resolve();
+      });
+    })
```

Comment:
```suggestion
    await new Promise(function(resolve, reject) {
      loadScript(url)
        .done(function() {
          scripts.push(url);
          return resolve();
        }).fail(function(_jqXHR, _textStatus, err) {
          scripts.push(url);
          console.log('getScript: Failed to load:', url, err);
          return resolve();
      });
    })
```

Reply:
So, surprisingly this seems to fix both problems, but I don't understand how exactly.

Reply:
Looks good - scripts is a list of the plugins that have been seen. Allows the loading to not repeat trying to load plugins that are not available.